api/METRICS-DASHBOARD.md

```diff
@@ -0,0 +1,73 @@
+# Proposal for metrics/dashboard API
+
+Goals:
+* Unified treatment of many different kinds of metrics in dashboard
+* A clean separation between dashboard (for visualization) and different kinds of backends (for metric calculation)
+
+Status quo:
+* Dashboard currently keeps track of the metric values in the cache dictionary `fairlearn.widget._fairlearn_widget.FairlearnWidget._response`
+* The dictionary is updated in `fairlearn.widget._fairlearn_dashboard.FairlearnDashboard._on_request`
+* There's a PR out to fill the whole data structure in  `fairlearn.metrics.create_dashboard_dictionary`
+
+Issues with the status quo:
+* Code duplication / redundancy / brittleness due to copy-paste errors
+* Currently only one kind of a metric (`<metric>_summary`) is supported and hard-wired into the dashboard dictionary
+
+## Proposal
+
+### Part I: More general dashboard dictionary
+
+```
+{
+  "prediction_type":  "binary_classification" or "probabilistic_binary_classification" or "regression",
+  "y_true": float[],
+  "y_pred": float[][],

```

Comment:
If we change the keys, that comes under the heading of Major Change at this point.

Reply:
If we're doing that, then we should merge `modelNames` and `predictedY` into a list of dictionaries, similar to the `sensitive_features`.

Reply:
yes... part of a new PR.